test/Tmds.Ssh.Tests/PrivateKeyCredentialTests.cs

```diff
@@ -155,6 +163,52 @@ await RunWithKeyConversion(_sshServer.TestUserIdentityFileEcdsa521, async (strin
         }, async (c) => await c.ConnectAsync());
     }
 
+    [Fact]
+    public async Task Ecdsa256InMemoryKey()
+    {
+        Name expectedalgorithm = new Name("ecdsa-sha2-nistp256"u8.ToArray());
+
+        var credential = new ECDsaKeyCredential(ECCurve.NamedCurves.nistP256);
+        using var privateKey = await credential.LoadKeyAsync(default);
+        Assert.NotNull(privateKey);
+        Assert.Single(privateKey.Algorithms);
+        Assert.Equal(expectedalgorithm, privateKey.Algorithms[0]);
+    }
+
+    [Fact]
+    public async Task Ecdsa384InMemoryKey()
+    {
+        Name expectedalgorithm = new Name("ecdsa-sha2-nistp384"u8.ToArray());
+
+        var credential = new ECDsaKeyCredential(ECCurve.NamedCurves.nistP384);
+        using var privateKey = await credential.LoadKeyAsync(default);
+        Assert.NotNull(privateKey);
+        Assert.Single(privateKey.Algorithms);
+        Assert.Equal(expectedalgorithm, privateKey.Algorithms[0]);
+    }
+
+    [Fact]
+    public async Task Ecdsa521InMemoryKey()
+    {
+        Name expectedalgorithm = new Name("ecdsa-sha2-nistp521"u8.ToArray());
+
+        var credential = new ECDsaKeyCredential(ECCurve.NamedCurves.nistP521);
+        using var privateKey = await credential.LoadKeyAsync(default);
+        Assert.NotNull(privateKey);
+        Assert.Single(privateKey.Algorithms);
+        Assert.Equal(expectedalgorithm, privateKey.Algorithms[0]);
+    }
+
+    [Fact]
+    public async Task FailEcdsaKeyWithInvalidCurve()
+    {
+        const string expected = "Curve 'brainpoolP256r1' is not known.";
+
+        var credential = new ECDsaKeyCredential(ECCurve.NamedCurves.brainpoolP256r1);
+        NotSupportedException exc = await Assert.ThrowsAsync<NotSupportedException>(async () => await credential.LoadKeyAsync(default));
+        Assert.Equal(expected, exc.Message);
+    }
```

Comment:
Good tests!

side-note: I'd like us to update the existing tests in this class to speed it up by: pre-generating keys and storing them as const strings (like the existing `PrivateRsaKey`) and trying to load them and check values (instead of doing a full connect).

Reply:
I can certainly update these new tests to load the key from a serialized string. This particular test doesn't do a full connection and while the others do so I think there is some merit for actually testing the end to end connection. Ultimately it's up to you whether you prefer the shorter and quicker test setup which certainly has its merits.

Reply:
The tests added here are good.

> Ultimately it's up to you whether you prefer the shorter and quicker test setup which certainly has its merits.

Yes, I'd like to put some effort into making the tests run faster (without sacrificing coverage).